node_modules/oae-principals/tests/test-dao.js

```diff
@@ -228,5 +217,59 @@ describe('Principals DAO', function() {
                 });
             });
         });
+
+        /**
+         * Test that verifies created user appears in PrincipalsDAO.iterateAll
+         */
+        it('verify newly created principals are returned in iterateAll', function(callback) {
+            // Create a user to test with
+            TestsUtil.generateTestUsers(camAdminRestContext, 1, function(err, user) {
+                assert.ok(!err);
+                user = _.values(user)[0];
+
+                var testUsername = TestsUtil.generateTestUserId();
+
+                // Stores how many principals were in the database before we created a new one
+                var numPrincipalsOrig = 0;
+
+                // Count how many principals we currently have in the database
+                PrincipalsDAO.iterateAll(null, 1000, function(principalRows, done) {
+                    if (principalRows) {
+                        numPrincipalsOrig += principalRows.length;
+                    }
+
+                    return done();
+                }, function(err) {
+                    assert.ok(!err);
+
+                    // Create one new one, and ensure the new number of principals is numPrincipalsOrig + 1
+                    TestsUtil.generateTestUsers(camAdminRestContext, 1, function(err, users, createdUser) {
+                        assert.ok(!err);
+
+                        var numPrincipalsAfter = 0;
+                        var hasNewPrincipal = false;
+
+                        // Count the principals we have now, and ensure we iterate over the new user
+                        PrincipalsDAO.iterateAll(null, 1000, function(principalRows, done) {
+                            if (principalRows) {
+                                numPrincipalsAfter += principalRows.length;
+                                _.each(principalRows, function(principalRow) {
+                                    if (principalRow.principalId === createdUser.user.id) {
+                                        hasNewPrincipal = true;
+                                    }
+                                });
+                            }
+
+                            return done();
+                        }, function(err) {
+                            assert.ok(!err);
+                            assert.strictEqual(numPrincipalsOrig + 1, numPrincipalsAfter);
+                            assert.ok(hasNewPrincipal);
+                            return callback();
+                        });
+                    });
+                });
+            });
+        });
```

Comment:
Unfortunately, it butchers the diff pretty good. But if you look at the `verify updating email address retains duplicate email mappings` test on its own, that's basically all that's being added here.